src/sentry/snuba/discover.py

```diff
@@ -136,6 +136,61 @@ def create_reference_event_conditions(reference_event):
     return conditions
 
 
+# TODO (evanh) This whole function is here because we are using the max value to
+# calculate the entire bucket width. If we could do that in a smarter way,
+# we could avoid this whole calculation.
+def find_histogram_buckets(field, params):
+    match = is_function(field)
+    if not match:
+        raise InvalidSearchQuery(u"received {}, expected histogram function".format(field))
+
+    columns = [c.strip() for c in match.group("columns").split(",") if len(c.strip()) > 0]
+
+    if len(columns) != 2:
+        raise InvalidSearchQuery(
+            u"histogram(...) expects 2 column arguments, received {:g} arguments".format(
+                len(columns)
+            )
+        )
+
+    column = columns[0]
+    # TODO evanh: This can be expanded to more fields at a later date, for now keep this limited.
+    if column != "transaction.duration":
+        raise InvalidSearchQuery(
+            "histogram(...) can only be used with the transaction.duration column"
+        )
```

Comment:
Should we take a parameter if it only has one valid value? Could histograms be generated for other numeric columns in theory?

Reply:
Yep definitely. As long as the column is numeric you can bucket it.